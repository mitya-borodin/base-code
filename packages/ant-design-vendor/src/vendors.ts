import "antd";
import "antd/dist/antd.css";
import "antd/es/select";
import "antd/lib/locale-provider/en_US";
import "antd/lib/table";
