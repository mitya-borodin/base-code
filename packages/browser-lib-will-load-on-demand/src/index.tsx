import "./index.css";

console.log("Entry point for launch in browser");
