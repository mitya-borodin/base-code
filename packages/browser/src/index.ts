// ENUMS
export { repositoryPubSubEnum } from "./enums/repositoryPubSubEnum";
export { SingleRepositoryPubSubEnum } from "./enums/SingleRepositoryPubSubEnum";
// FILTER
export { FilterStore } from "./filter/FilterStore";
// FORM
export { FormStore } from "./form/FormStore";
export { RepositoryFormStore } from "./form/RepositoryFormStore";
export { SingleFormStore } from "./form/SingleFormStore";
export { CacheRepository } from "./repository/CacheRepository";
export { Repository } from "./repository/Repository";
// REPOSITORY
export { SingleObjectRepository } from "./repository/SingleObjectRepository";
// TRANSPORTS
export { BaseHttpTransport } from "./transport/http/BaseHttpTransport";
export { RepositoryHttpTransport } from "./transport/http/RepositoryHttpTransport";
export { SingleObjectHttpTransport } from "./transport/http/SingleObjectHttpTransport";
export { WSClient } from "./transport/ws/WSClient";
// USER
export { UserHTTPTransport } from "./user/UserHTTPTransport";
export { UserRepository } from "./user/UserRepository";
