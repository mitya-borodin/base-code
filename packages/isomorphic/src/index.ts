export { DataTransferObject, ValueObject, Entity, EntityId } from "./Entity";

export { userEventEnum } from "./user/userEventEnum";
export { userGroupEnum } from "./user/userGroupEnum";
export { User, UserData } from "./user/User";

export { Log, LogData } from "./log/Log";
export { LogType } from "./log/LogType";
export { logEnum } from "./log/logEnum";
export { logTypeEnum } from "./log/logTypeEnum";

export { Validation, ValidationData } from "./validation/Validation";
export { ValidationResult } from "./validation/ValidationResult";

export {
  ErrorChannel,
  PingChannel,
  PongChannel,
  BindUserToConnection,
  UnbindUserFromConnection,
} from "./webSocket/wsChannels";
export { wsEventEnum } from "./webSocket/wsEventEnum";
export { makeErrorMessage, makeMessage, recognizeMessage } from "./webSocket/wsHelpers";
export { Send } from "./webSocket/Send";

export { ListResponse, Response } from "./Response";
