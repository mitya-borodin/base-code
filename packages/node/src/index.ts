// Application
export {
  getAuthenticateMiddleware,
  getAuthenticateStrategyMiddleware,
  setCookieForAuthenticate,
  unsetCookieForAuthenticate,
} from "./app/auth";
export { Config } from "./app/Config";
export { getRequestBodyJson } from "./app/getRequestBodyJson";
export { KoaServer } from "./app/KoaServer";
// Model
export { Model } from "./model/Model";
export { MongoDBConnection } from "./model/MongoDBConnection";
export { MongoDBRepository } from "./model/MongoDBRepository";
export { SingleObjectModel } from "./model/SingleObjectModel";
export { UserModel } from "./model/UserModel";
// Transport
export { BaseHttpTransport, BaseHttpTransportACL } from "./transport/BaseHttpTransport";
export { downloadFile } from "./transport/downloadFile";
export { HttpTransport, HttpTransportACL } from "./transport/HttpTransport";
export {
  SingleObjectHttpTransport,
  SingleObjectHttpTransportACL,
} from "./transport/SingleObjectHttpTransport";
export { uploadFile } from "./transport/uploadFile";
export { UserHttpTransport, UserHttpTransportACL } from "./transport/UserHttpTransport";
// UTILS
export { authenticate } from "./utils/authenticate";
export { encryptPassword } from "./utils/encryptPassword";
export { getSalt } from "./utils/getSalt";
// WEB_SOCKET
export { Channels } from "./webSocket/Channels";
export { Connection } from "./webSocket/Connection";
export { WebSocketServer } from "./webSocket/WebSocketServer";
