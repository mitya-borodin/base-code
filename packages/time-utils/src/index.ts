export { dateOnly } from "./dateOnly";
export { formatMinutes } from "./formatMinutes";
export { intersection } from "./intersection";
export { minutesToTimeOnly } from "./minutesToTimeOnly";
export { overlaps } from "./overlaps";
export { overlapsRS } from "./overlapsRS";
export { separateMinutes } from "./separateMinutes";
export { timeOnly } from "./timeOnly";
export { yearOnly } from "./yearOnly";
