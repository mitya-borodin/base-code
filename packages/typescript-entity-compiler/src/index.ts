export function compiler(): void {
  console.log(1111);
}
