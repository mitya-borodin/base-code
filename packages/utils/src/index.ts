export { checkPassword } from "./checkPassword";
export { getAttribute } from "./getAttribute";
export { getErrorMessage } from "./getErrorMessage";
export {
  isArray,
  isBoolean,
  isDate,
  isFunction,
  isNumber,
  isObject,
  isString,
  isUndefined,
  isNull,
} from "./isType";
export { uploadCSV } from "./uploadCSV";
export { uploadHTML } from "./uploadHTML";
