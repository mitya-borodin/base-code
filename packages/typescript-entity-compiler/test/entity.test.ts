import { compiler } from "../src";

describe("Compilation for ", () => {
  it("a new type declaration", () => {
    expect(compiler()).toMatchSnapshot();
  });
  it("already compiled type declaration without additional code of developer", () => {
    expect(compiler()).toMatchSnapshot();
  });
  it("already compiled type declaration with additional code which was wrote by developers", () => {
    expect(compiler()).toMatchSnapshot();
  });
});
