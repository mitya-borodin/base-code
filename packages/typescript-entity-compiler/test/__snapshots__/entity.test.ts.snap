// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compilation for a new type declaration 1`] = `undefined`;

exports[`Compilation for already compiled type declaration with additional code which was wrote by developers 1`] = `undefined`;

exports[`Compilation for already compiled type declaration without additional code of developer 1`] = `undefined`;
